dotCMS/src/main/java/com/dotcms/storage/FileSystemStorage.java

```diff
@@ -0,0 +1,260 @@
+package com.dotcms.storage;
+
+import com.dotcms.concurrent.DotConcurrentFactory;
+import com.dotmarketing.exception.DotRuntimeException;
+import com.dotmarketing.util.Config;
+import com.dotmarketing.util.Logger;
+import com.google.common.collect.ImmutableList;
+import com.liferay.util.FileUtil;
+import org.apache.commons.io.FileUtils;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.Future;
+
+public class FileSystemStorage implements Storage {
+
+    private final Map<String, File> buckets = new ConcurrentHashMap<>();
+
+    /**
+     * Adds a mapping between a bucket name and a file
+     * @param bucketName {@link String} bucket name
+     * @param file {@link File}
+     */
+    public void addBucketMapping (final String bucketName, final File file) {
+
+        this.buckets.put(bucketName, file);
+    }
+
+    @Override
+    public boolean existsBucket(final String bucketName) {
+
+        return this.buckets.containsKey(bucketName) && this.buckets.get(bucketName).exists();
+    }
+
+    @Override
+    public boolean existsObject(final String bucket, final String objectPath) {
+
+        return this.existsBucket(bucket) && new File(this.buckets.get(bucket), objectPath).exists();
+    }
+
+    @Override
+    public boolean createBucket(final String bucketName) {
+
+        throw new UnsupportedOperationException("On FileSystemStorage can not create buckets, they have to be previously defined");
+    }
+
+    @Override
+    public boolean createBucket(final String bucketName, final Map<String, Object> extraOptions) {
+
+        throw new UnsupportedOperationException("On FileSystemStorage can not create buckets, they have to be previously defined");
+    }
+
+    @Override
+    public boolean deleteBucket(final String bucketName) {
+
+        throw new UnsupportedOperationException("On FileSystemStorage can not delete buckets");
+    }
+
+    public boolean deleteObject(final String bucket, final String path) {
+
+        return new File(this.buckets.get(bucket), path).delete();
+    }
+
+    @Override
+    public List<Object> listBuckets() {
+
+        return new ImmutableList.Builder<>().addAll(this.buckets.keySet()).build();
+    }
+
+    @Override
+    public Object pushFile(final String bucketName,
+                       final String path,
+                       final File file,
+                       final Map<String, Object> extraMeta) {
+
+        if (!this.existsBucket(bucketName)) {
+
+            throw new IllegalArgumentException("The bucketName: " + bucketName +
+                    ", does not have any file mapped");
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [The String literal ", does not have any file mapped" appears 4 times in this file; the first occurrence is on line 84](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5547873)